cmd/talosctl/cmd/talos/config.go

```diff
@@ -186,6 +188,106 @@ var configAddCmd = &cobra.Command{
 	},
 }
 
+// configRemoveCmdFlags represents the `config remove` command flags.
+var configRemoveCmdFlags struct {
+	noconfirm bool
+	dry       bool
+}
+
+// configRemoveCmd represents the `config remove` command.
+var configRemoveCmd = &cobra.Command{
+	Use:   "remove <context>",
+	Short: "Remove a context",
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		pattern := args[0]
+		if pattern == "" {
+			return fmt.Errorf("no context specified")
+		}
+
+		c, err := clientconfig.Open(GlobalArgs.Talosconfig)
+		if err != nil {
+			return fmt.Errorf("error reading config: %w", err)
+		}
+
+		if len(c.Contexts) == 0 {
+			return errors.New("no contexts defined")
+		}
+
+		matches := maps.Keys(
+			maps.Filter(c.Contexts, func(context string, _ *clientconfig.Context) bool { return glob.Glob(pattern, context) }),
+		)
+		if len(matches) == 0 {
+			return fmt.Errorf("no contexts matched %q", pattern)
+		}
+
+		sort.Slice(matches, func(i, j int) bool { return matches[i] < matches[j] })
+
+		if len(c.Contexts) == 1 && matches[0] == c.Context {
+			if !configRemoveCmdFlags.noconfirm {
+				promt := fmt.Sprintf("Are you sure you want to remove the only %q context?", c.Context)
+
+				if !helpers.Confirm(promt) || configRemoveCmdFlags.dry {
+					return nil
+				}
+			}
+
+			return removeConfig(c)
+		}
+
+		noChanges := true
+
+		for _, match := range matches {
+			if match == c.Context {
+				fmt.Fprintf(os.Stderr, "Skipping removal of the current context %q, please change it to another before removing\n", match)
```

Comment:
Lets leave this decision to the user. The subtle '*' succeeding the current context acts as a warning in itself.

Reply:
I need confirmation on this one before pushing new changes. How do you feel about this @DmitriyMV ?

Reply:
Sorry for the delay. We don't want to remove the current context, because `talosctl` it starts to produce errors like `error constructing client: failed to determine endpoints` which is not exactly descriptive. Granted, it will produce the same error message if there is no config at all, but that is at least easier to debug.

We can do two things. Either:
a) Do not allow the removal of the current context.
b) Switch to another context (which one?).

But we cannot leave current context blank if other contexts are present.

Reply:
Did not see your changes and forced pushed my own, sorry for that.

Skipping the current context (option a) is a much better option rather than switching to another one. Also the following code will fail if all the contexts have already been deleted.

```go
		if !noChanges && c.Context == "" && len(c.Contexts) > 0 {
			// we ensure that there is always a current context if there
			// are any contexts

			// sortInPlace receives `nil` if len(c.Contexts) == 0
			c.Context = sortInPlace(maps.Keys(c.Contexts))[0]
		}
```